text/metrics/0238-pipeline-monitoring.md

```diff
@@ -0,0 +1,426 @@
+`batch/error` and `batch/noerror`.
+
+#### Pipeline monitoring diagram
+
+The relationship between items received, dropped, and exported is
+shown in the following diagram.
+
+![pipeline monitoring metrics](../images/otel-pipeline-monitoring.png)
+
+### Proposed metrics semantic conventions
+
+The proposed metric names match the following pattern:
+
+| Metric Name               | Meaning                                                            |
+|---------------------------|--------------------------------------------------------------------|
+| `otel.{station}.received` | Inclusive count of items entering the pipeline at a station.       |
+| `otel.{station}.dropped`  | Non-inclusive count of items dropped by a processor in a pipeline. |
+| `otel.{station}.exported` | Inclusive count of items exiting the pipeline at a station.        |
+
+The behavior specified for SDKs and Collectors at each level of detail
+is different, because SDKs do not receive items from a pipeline and
+because they outnumber the other components.
+
+These attributes can be applied to any of the pipeline monitoring
+metrics specified here.
+
+| Attributes     | Meaning                                     | Level of detail (Optional) | Examples                                                   |
+|----------------|---------------------------------------------|----------------------------|------------------------------------------------------------|
+| `otel.signal`  | Name of the telemetry signal                | Basic (Opt-out)            | `traces`, `logs`, `metrics`                                |
+| `otel.name`    | Type, name, or "type/name" of the component | Normal (Opt-out)           | `probabilitysampler`, `batch`, `otlp/grpc`                |
+| `otel.success` | Boolean: item considered success?           | Normal (Opt-out)           | `true`, `false`                                            |
+| `otel.reason`  | Explaination of success/failures.           | Detailed (Opt-in)          | `ok`, `timeout`, `permission_denied`, `resource_exhausted` |
+| `otel.scope`   | Name of instrumentation.                    | Detailed (Opt-in)          | `opentelemetry.io/library`                                 |
+
+For example, when a sampler processor drops an item it may report
+`success=true`, but when a queue processor drops an item it may report
+`success=false`.
+
+#### SDK default configuration
+
+| Metric name         | Enablement level |
+|---------------------|------------------|
+| `otel.sdk.received` | Basic            |
+| `otel.sdk.dropped`  | Normal           |
+| `otel.sdk.exported` | Detailed         |
+
+#### Collector default configuration
+
+| Metric name               | Enablement level |
+|---------------------------|------------------|
+| `otel.{station}.received` | Basic            |
+| `otel.{station}.dropped`  | Basic            |
+| `otel.{station}.exported` | Basic            |
```

Comment:
How would the `{station}` be set? A collector instance doesn't have that information. Will it be a mandatory configuration option?

What if we keep `otel.collector.(received|dropped|exported)` metric names and have another optional resource attribute to identify a station?

Reply:
The station concept was firmly rejected. This proposal is out of date, thanks.